src/Data/Managers/GetInstanceResources/QueryManager.test.tsx

```diff
@@ -0,0 +1,275 @@
+    version: 1,
+  });
+  return (
+    <RemoteDataView
+      label="Dummy"
+      data={data}
+      SuccessView={() => <div aria-label="Dummy-Success">success</div>}
+    />
+  );
+};
+
+test("Given the InstanceResourcesQueryManager When initial request fails with 409 Then requests are retried", async () => {
+  const { component, apiHelper, resourcesRequest, instanceRequest, resolveAs } =
+    setup();
+  render(component);
+
+  expect(apiHelper.pendingRequests).toEqual([resourcesRequest(1)]);
+  await act(resolveAs.resourcesConflict);
+  expect(
+    await screen.findByRole("generic", { name: "Dummy-Loading" })
+  ).toBeVisible();
+
+  expect(apiHelper.pendingRequests).toEqual([instanceRequest()]);
+  await act(resolveAs.instanceSuccess(2));
+  expect(apiHelper.pendingRequests).toEqual([resourcesRequest(2)]);
+  await act(resolveAs.resourcesSuccess);
+  expect(
+    await screen.findByRole("generic", { name: "Dummy-Success" })
+  ).toBeVisible();
+});
+
+test("Given the InstanceResourcesQueryManager When instance fails Then error is shown", async () => {
+  const { component, apiHelper, resourcesRequest, instanceRequest, resolveAs } =
+    setup();
+  render(component);
+
+  expect(apiHelper.pendingRequests).toEqual([resourcesRequest(1)]);
+  await act(resolveAs.resourcesConflict);
+  expect(
+    await screen.findByRole("generic", { name: "Dummy-Loading" })
+  ).toBeVisible();
+
+  expect(apiHelper.pendingRequests).toEqual([instanceRequest()]);
+  await act(resolveAs.error);
+  expect(
+    await screen.findByRole("generic", { name: "Dummy-Failed" })
+  ).toBeVisible();
+});
+
+test("Given the InstanceResourcesQueryManager When it keeps failing Then it stops at the retryLimit", async () => {
+  const { component, apiHelper, resourcesRequest, instanceRequest, resolveAs } =
+    setup();
+  render(component);
+
+  // 1st resources
+  expect(apiHelper.pendingRequests).toEqual([resourcesRequest(1)]);
+  await act(resolveAs.resourcesConflict);
+  expect(
+    await screen.findByRole("generic", { name: "Dummy-Loading" })
+  ).toBeVisible();
+
+  // 1st instance
+  expect(apiHelper.pendingRequests).toEqual([instanceRequest()]);
+  await act(resolveAs.instanceSuccess(2));
+
+  // 2nd resources
+  expect(apiHelper.pendingRequests).toEqual([resourcesRequest(2)]);
+  await act(resolveAs.resourcesConflict);
+
+  expect(
+    await screen.findByRole("generic", { name: "Dummy-Loading" })
+  ).toBeVisible();
+
+  // 2nd instance
+  expect(apiHelper.pendingRequests).toEqual([instanceRequest()]);
+  await act(resolveAs.instanceSuccess(3));
+
+  // 3rd resources
+  expect(apiHelper.pendingRequests).toEqual([resourcesRequest(3)]);
+  await act(resolveAs.resourcesConflict);
+  expect(
+    await screen.findByRole("generic", { name: "Dummy-Failed" })
+  ).toBeVisible();
+  expect(apiHelper.pendingRequests).toHaveLength(0);
+});
+
+test("Given the InstanceResourcesQueryManager Then a task is registered on the scheduler", async () => {
```

Comment:
The task is now always registered. The test is about handling a conflict in the registered task